datahub-web-react/src/app/search/filters/utils.tsx

```diff
@@ -0,0 +1,297 @@
+    } else if (filterEntity) {
+        if (entityRegistry.hasEntity(filterEntity.type)) {
+            icon = entityRegistry.getIcon(filterEntity.type, size || 12, IconStyleType.ACCENT, ANTD_GRAY[9]);
+            label = entityRegistry.getDisplayName(filterEntity.type, filterEntity);
+        } else if (filterEntity.type === EntityType.DataPlatformInstance) {
+            const { icon: newIcon, label: newLabel } = getDataPlatformInstanceIconAndLabel(
+                filterEntity,
+                entityRegistry,
+                size,
+            );
+            icon = newIcon;
+            label = newLabel;
+        } else {
+            label = filterValue;
+        }
+    } else {
+        label = filterValue;
+    }
+
+    if (filterLabelOverride) {
+        label = filterLabelOverride;
+    }
+
+    return { icon, label };
+}
+
+export function getNumActiveFiltersForFilter(activeFilters: FacetFilterInput[], filter: FacetMetadata) {
+    return activeFilters.find((f) => f.field === filter.field)?.values?.length || 0;
+}
+
+export function getNumActiveFiltersForGroupOfFilters(activeFilters: FacetFilterInput[], filters: FacetMetadata[]) {
+    let numActiveFilters = 0;
+    filters.forEach((filter) => {
+        numActiveFilters += getNumActiveFiltersForFilter(activeFilters, filter);
+    });
+    return numActiveFilters;
+}
+
+// combine existing aggs from search response with new aggregateAcrossEntities response
+export function combineAggregations(
+    filterField: string,
+    originalAggs: AggregationMetadata[],
+    newFacets?: FacetMetadata[] | null,
+) {
+    const combinedAggregations = [...originalAggs];
+    if (newFacets) {
+        const newAggregations = newFacets.find((facet) => facet.field === filterField)?.aggregations;
+        newAggregations?.forEach((agg) => {
+            // only add the new aggregations if it is not already in the original to avoid dupes
+            if (!originalAggs.find((existingAgg) => existingAgg.value === agg.value)) {
+                combinedAggregations.push(agg);
+            }
+        });
+    }
+    return combinedAggregations;
+}
+
+// filter out any aggregations with a count of 0 unless it's already selected and in activeFilters
+export function filterEmptyAggregations(aggregations: AggregationMetadata[], activeFilters: FacetFilterInput[]) {
+    return aggregations.filter((agg) => {
+        if (agg.count === 0) {
+            return activeFilters.find((activeFilter) => activeFilter.values?.includes(agg.value));
+        }
+        return true;
+    });
+}
+
+export function sortFacets(facetA: FacetMetadata, facetB: FacetMetadata, sortedFacetFields: string[]) {
+    if (sortedFacetFields.indexOf(facetA.field) === -1) return 1;
+    if (sortedFacetFields.indexOf(facetB.field) === -1) return -1;
+    return sortedFacetFields.indexOf(facetA.field) - sortedFacetFields.indexOf(facetB.field);
+}
+
+export function getFilterDropdownIcon(field: string) {
```

Comment:
nice! 

i think it will be useful to have the series of steps required to add a new filter to the new search filtering experience. ideally not too many!

Reply:
yeah! honestly it should be real simple. the only real special logic is adding a nice icon for the filter. otherwise it just won't have an icon on the filter dropdown, so it's not like new filters won't work, they just may not be as pretty.